### PR TITLE
[jsonlogencodingextension] Move to alpha

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Move jsonlogencodingextension to alpha

### CODE REVIEW

**Review by MovieStoreGuy — APPROVED** [review]

File: .chloggen/alpha_jsonlog.yaml
Comment (crobert-1) [inline .chloggen/alpha_jsonlog.yaml:4]:
```suggestion
change_type: new_component
```
Since this isn't being shipped with contrib yet, I assume you're planning on adding it to the contrib distribution? If so, we can say mark it as a new component, unless I'm missing something.

**Review by crobert-1 — APPROVED** [review]

File: .chloggen/alpha_jsonlog.yaml
Comment (atoulme) [reply .chloggen/alpha_jsonlog.yaml:4]: I am not sure how much that matters. This is just going to a different section of CHANGELOG.

Comment (crobert-1) [reply .chloggen/alpha_jsonlog.yaml:4]: Sure, but that's the point of these fields right? It's not a big deal, I just think this change makes more sense being labelled as a new component vs. enhancement 🤷

**Review by codeboten — COMMENTED** [review]
@crobert-1 can you confirm if your comment has been resolved? 

@atoulme please resolve the conflict 👍🏻

File: .chloggen/alpha_jsonlog.yaml
Comment (crobert-1) [reply .chloggen/alpha_jsonlog.yaml:4]: I'll leave it up to you, I saw a lot of these extensions were moving to `alpha`, so I realize this was part of a larger effort 👍

**Review by crobert-1 — APPROVED** [review]

**Review by dao-jun — APPROVED** [review]

**Comment by atoulme** [comment]
Conflict resolved.

**Comment by crobert-1** [comment]
`Generated code is out of date, please run "make genotelcontribcol" and commit the changes in this PR.`

**Comment by atoulme** [comment]
Ready to merge - @open-telemetry/collector-contrib-maintainer please review

**Review by TylerHelmuth — APPROVED** [review]

**Review by evan-bradley — APPROVED** [review]

**Pull request merged**
